valohai_yaml/lint.py

```diff
@@ -65,6 +65,53 @@ def lint_file(file_path: str) -> LintResult:
             return lr
 
 
+def override_validator(data: List) -> Generator[ValidationError, Any, None]:
+    pipeline_dict, step_dict = get_validation_data(data)
+
+    for pipeline, pipeline_values in pipeline_dict.items():
+        for node_name, nodes in pipeline_values.items():
+            for step_key, node_value in nodes.items():
+                step = step_dict.get(step_key)
+                if step:
+                    inputs = node_value.get('inputs', [])
+                    parameters = node_value.get('parameters', [])
+                    for item, category in [(inputs, 'inputs'), (parameters, 'parameters')]:
+                        for element in item:
+                            element_name = element['name']
+                            if element_name not in step[category]:
+                                yield ValidationError(
+                                    f"{category[:-1]} '{element_name}' overridden in node '{node_name}' "
+                                    f"not found in step '{step_key}'")
+                else:
+                    yield ValidationError(f"step '{step_key}' not found in steps for '{pipeline}'")
+
+
+def get_validation_data(data: List) -> Tuple:
+    step_dict = {}
+    pipeline_dict = {}
+    for step in data:
+        step_data = step.get('step')
+        if step_data:
+            step_name = step_data.get('name')
+            if step_name:
+                step_dict[step_name] = {
+                    'inputs': [item['name'] for item in step_data.get('inputs', [])],
+                    'parameters': [item['name'] for item in step_data.get('parameters', [])]
+                }
+
+        pipeline_data = step.get('pipeline')
+        if pipeline_data:
+            pipeline_name = pipeline_data.get('name')
+            if pipeline_name:
+                pipeline_nodes = pipeline_data.get('nodes', [])
+                pipeline_dict[pipeline_name] = {
+                    i['name']: {i['step']: i.get('override', {})}
+                    for i in pipeline_nodes if 'step' in i
+                }
+
+    return pipeline_dict, step_dict
```

Comment:
Ah, I think this validation (if it does what I think it does) should maybe be done in `Pipeline`'s `lint()`, where everything has already been parsed. You can access the entirety of the config via `context["config"]` there.

Let me know if I've misunderstood what's going on here...